src/bootstrap/config.rs

```diff
@@ -1519,18 +1516,26 @@ impl Config {
             config.llvm.build_config = llvm.build_config.clone().unwrap_or(Default::default());
 
             let asserts = llvm_assertions.unwrap_or(false);
-            config.llvm.from_ci = match llvm.download_ci_llvm {
-                Some(StringOrBool::String(s)) => {
-                    assert!(s == "if-available", "unknown option `{}` for download-ci-llvm", s);
-                    crate::llvm::is_ci_llvm_available(&config, asserts)
-                }
+            let mut downloaded_config = false;
+            config.llvm_from_ci = match llvm.download_ci_llvm {
+                Some(StringOrBool::String(s)) => match s.as_str() {
+                    "if-available" => crate::llvm::is_ci_llvm_available(&config, asserts),
+                    "if-identical" => match crate::llvm::get_llvm_opts_from_ci(&config) {
```

Comment:
this should imply `if-available` IMO, otherwise you'll get the errors from before where it'll try to download even if it was changed in the current commit (as well as issues on tier 2/3 platforms)
```suggestion
                    "if-identical" => crate::llvm::is_ci_llvm_available(&config, asserts) && match crate::llvm::get_llvm_opts_from_ci(&config) {